src/main/java/io/candydoc/ddd/model/FoundDomainObjects.java

```diff
@@ -0,0 +1,115 @@
+package io.candydoc.ddd.model;
+
+import io.candydoc.ddd.extract_domain_vocabulary.WrongUseOfDDDPatterns;
+import io.candydoc.ddd.strategic_pattern.BoundedContext;
+import io.candydoc.ddd.strategic_pattern.SharedKernel;
+import io.candydoc.ddd.tactical_pattern.Aggregate;
+import io.candydoc.ddd.tactical_pattern.DomainCommand;
+import io.candydoc.ddd.tactical_pattern.DomainEvent;
+import io.candydoc.ddd.tactical_pattern.ValueObject;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@EqualsAndHashCode
+@ToString
+public final class FoundDomainObjects {
+
+  private final Map<CanonicalName, FoundDomainObject> domainObjects = new HashMap<>();
+
+  public FoundDomainObjects add(FoundDomainObject domainObject) {
+    domainObjects.put(domainObject.canonicalName(), domainObject);
+    return this;
+  }
+
+  public FoundDomainObjects addAll(Collection<FoundDomainObject> domainObjects) {
+    domainObjects.forEach(this::add);
+    return this;
+  }
+
+  public boolean isEmpty() {
+    return domainObjects.isEmpty();
+  }
+
+  public void checkConsistency() {
+    Set<FoundDomainObject> invalidDomainModels =
+        domainObjects.values().stream()
+            .filter(FoundDomainObject::matchMoreThanOneDDDPattern)
+            .collect(Collectors.toUnmodifiableSet());
+
+    if (!invalidDomainModels.isEmpty()) {
+      throw new WrongUseOfDDDPatterns(invalidDomainModels);
+    }
+  }
+
+  public Set<DomainModel> toDomainModels() {
+    return domainObjects.values().stream()
+        .map(this::toDomainModel)
+        .collect(Collectors.toUnmodifiableSet());
+  }
+
+  private DomainModel toDomainModel(FoundDomainObject foundDomainObject) {
```

Comment:
Should be in FoundDomainObject

Reply:
It's not possible because the returned DomainModel handle the dependencies found thanks to other found objects. You need all the found concepts to create the rights DomainModel objects (line 57 and line 64)